iOS/Flows/Conversation/Cells/TimeMachineCollectionViewLayout.swift

```diff
@@ -318,11 +359,15 @@ class TimeMachineCollectionViewLayout: UICollectionViewLayout {
         var frontmostIndexes: [IndexPath] = []
         for i in 0..<sectionCount {
             guard let frontmostIndex = self.getFrontmostIndexPath(in: i) else { continue }
-            frontmostIndexes.append(frontmostIndex)
+            guard let range = self.zRangesDict[frontmostIndex] else { continue }
+
+            if range.vector(to: self.zPosition) > -self.itemHeight {
+                frontmostIndexes.append(frontmostIndex)
+            }
```

Comment:
Fixes an issue where the newest message wasn't highlighted for certain edge cases.